### PR TITLE
feat(components): 'Editable' passes through html props. [Canary]

### PR DESCRIPTION
**Pull request by wodenx**
## Changes
- Allow the bodiless "Editable" component to receive html props and pass them on to the span it renders.
- Moves node key for vital copyright row up to footer.

## Test Instructions
- On the vital-demo site, verify that the hero card title has the following props:
  - `data-layer-region="Card:Title"`
  - `data-shadowed-by="__vitaltest__:Card:HeroLeftImageCentered"`
 - Validate that copyright renders correctly and is still editable

### CODE REVIEW

**Review by dewen — APPROVED** [review]

**Review by marcopagliarulo — APPROVED** [review]

File: packages/vital-layout/src/components/Footer/CopyrightRow/CopyrightRowClean.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ import type { CopyrightRowComponents, CopyrightRowProps } from './types';
 const copyrightRowComponents: CopyrightRowComponents = {
   Wrapper: Div,
   CopyrightWrapper: Div,
+  Disclaimer: Fragment,
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/vital-layout/src/components/Footer/CopyrightRow/CopyrightRowClean.tsx:26]: well both disclaimer and copyright are inside copyright wrapper

**Review by dewen — APPROVED** [review]

File: packages/bodiless-components/src/Editable.tsx [hunk 2]
```diff
@@ -59,21 +64,25 @@ export const isEditableData = (d: any): d is EditableData => {
   return true;
 };
 
-const Text = observer((props: EditableProps) => {
-  const { placeholder, useOverrides = () => ({}) }: EditableProps = props;
+const Text = observer((props: EditableBaseProps) => {
```
Comment (VancheeZze) [inline packages/bodiless-components/src/Editable.tsx:67]: Shouldn't it be `EditableProps` here as well?

**Review by VancheeZze — APPROVED** [review]

File: packages/bodiless-components/src/Editable.tsx [hunk 3]
```diff
@@ -59,21 +64,25 @@ export const isEditableData = (d: any): d is EditableData => {
...
+  }: EditableProps = props;
   const { sanitizer = identity }: EditableOverrides = useOverrides(props);
   const { node } = useNode<EditableData>();
   const text = sanitizer(
-    (node.data.text !== undefined ? node.data.text : props.children) || placeholder || '',
+    (node.data.text !== undefined ? node.data.text : children) || placeholder || '',
   );
   // eslint-disable-next-line react/no-danger
-  return <span dangerouslySetInnerHTML={{ __html: text }} />;
+  return <span {...rest} dangerouslySetInnerHTML={{ __html: text }} />;
```
Comment (VancheeZze) [inline packages/bodiless-components/src/Editable.tsx:77]: Also, just to double-check — the `nodeKey` prop does not slide into the span and there are no React warnings in the console regarding the incorrectly spelled property, right?

**Review by dewen — APPROVED** [review]

File: packages/bodiless-components/src/Editable.tsx (hunk 2 above)
Comment (wodenx) [reply packages/bodiless-components/src/Editable.tsx:66]: It should be EditableBasePrps everywhere - EditableProps adds the props which are added by wthNode

File: packages/bodiless-components/src/Editable.tsx (hunk 3 above)
Comment (wodenx) [reply packages/bodiless-components/src/Editable.tsx]: nodeKey is removed by withNode

**Review by VancheeZze — APPROVED** [review]

**Review by dewen — DISMISSED** [review]

**Review by dewen — APPROVED** [review]

**Review by VancheeZze — APPROVED** [review]

File: packages/vital-layout/src/components/Footer/tokens/vitalFooter.ts [hunk 4]
```diff
@@ -50,7 +50,7 @@ const Base = asFooterToken({
     Wrapper: vitalColor.BgSecondaryFooter,
   },
   Schema: {
-    FooterMenu: withNodeKey({ nodeKey: 'footer-menu', nodeCollection: 'site' }),
+    FooterMenu: withNodeKey('footer-menu'),
```
Comment (marcopagliarulo) [inline packages/vital-layout/src/components/Footer/tokens/vitalFooter.ts:53]: why removing the nodeCollection here?

Comment (wodenx) [reply packages/vital-layout/src/components/Footer/tokens/vitalFooter.ts:53]: bc the nodeCollextion is established in the Footer slot of the Layout.

**Review by marcopagliarulo — APPROVED** [review]

**Pull request merged**